server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -587,8 +550,8 @@ public void deleteStaleSegments(int lastNMetadataFilesToKeep) throws IOException
         }
 
         List<String> metadataFilesEligibleToDelete = sortedMetadataFileList.subList(
-            0,
-            sortedMetadataFileList.size() - lastNMetadataFilesToKeep
+            lastNMetadataFilesToKeep,
```

Comment:
Why is this needed now?

Reply:
we would need to delete all files except  the first `lastNMetadataFilesToKeep` .